### PR TITLE
Improve joins on mismatch and finish select table to join implementation

### PR DESCRIPTION
**Pull request by ahuang11**
Updates prompt so LLM knows to perform nearest join match.

image

On randomized data, it seems to work
image

Another example; runs a bit slow on closest match

image

image

---

This PR also fixes using NestedSelect instead of Upload to join table.

### CODE REVIEW

**Review by philippjfr — APPROVED** [review]

File: lumen/ai/prompts/sql_query.jinja2 [hunk 1]
```diff
@@ -9,16 +9,19 @@ The data for table '{{ table }}' follows the following JSON schema:
 {% if not join_required %}
 It was already determined that no join is required, so only use the existing
 table '{{ table }}' to calculate the result.
+{% else %}
+If the join's values do not align 50% of the time, then perform a join based on the
```
Comment (philippjfr) [inline lumen/ai/prompts/sql_query.jinja2:13]: How would it know whether it's likely to align?

Comment (ahuang11) [reply lumen/ai/prompts/sql_query.jinja2:13]: Potentially schema enums (I'll have to double check dates) or min/max

Comment (philippjfr) [reply lumen/ai/prompts/sql_query.jinja2:13]: Should be `min/max`.

Comment (ahuang11) [reply lumen/ai/prompts/sql_query.jinja2]: Seems to work with min/max
image

Comment (ahuang11) [reply lumen/ai/prompts/sql_query.jinja2]: Seems to work with string enums too 

image

Comment (ahuang11) [reply lumen/ai/prompts/sql_query.jinja2]: And lastly for numeric matching min/max

image

**Review by philippjfr — APPROVED** [review]

**Pull request merged**